scylla-cdc/src/consumer.rs

```diff
@@ -207,6 +219,21 @@ impl CDCRow<'_> {
         }
     }
 
+    /// Allows to take deleted elements from a collection.
+    /// Returns new empty vector if the value is null or such column doesn't exist.
+    /// The returned value is always owned.
+    /// Leaves None in place of taken data.
+    pub fn take_deleted_elements(&mut self, name: &str) -> Vec<CqlValue> {
+        let val = self.schema.deleted_el_mapping.get(name).map(|id| {
```

Comment:
If you change `map` to `and_then` here then you can get rid of flatten at the end.

You can also make the whole expression less nested:

```rust
self.schema
    .deleted_el_mapping
    .get(name)
    .and_then(|id| self.data[*id].take())
    .and_then(|x| match x {
        Set(x) => Some(x),
        _ => None,
    })
    .unwrap_or_default()
```